astropy/units/core.py

```diff
@@ -1623,64 +1623,52 @@ class EquivalentUnitsList(list):
             "Unit definition",
             "Aliases",
         )
-        # len(HEADING_NAMES), but hard-code since it is constant
-        ROW_LEN: Final[Literal[3]] = 3
         NO_EQUIV_UNITS_MSG: Final[str] = "There are no equivalent units"
 
         def __repr__(self) -> str:
-            if len(self) == 0:
+            if not self:
                 return self.NO_EQUIV_UNITS_MSG
-            else:
-                lines = self._process_equivalent_units(self)
-                lines.insert(0, self.HEADING_NAMES)
-                widths = [0] * self.ROW_LEN
-                for line in lines:
-                    for i, col in enumerate(line):
-                        widths[i] = max(widths[i], len(col))
-
-                f = "  {{0:<{}s}} | {{1:<{}s}} | {{2:<{}s}}".format(*widths)
-                lines = [f.format(*line) for line in lines]
-                lines = lines[0:1] + ["["] + [f"{x} ," for x in lines[1:]] + ["]"]
-                return "\n".join(lines)
+            lines = self._process_units()
+            widths = list(map(len, self.HEADING_NAMES))
+            for line in lines:
+                widths = [max(w, len(col)) for w, col in zip(widths, line, strict=True)]
+            row_template = "  {{0:<{}s}} | {{1:<{}s}} | {{2:<{}s}}".format(*widths)
+            return "\n".join(
+                [
+                    row_template.format(*self.HEADING_NAMES),
+                    "[",
+                    *(f"{row_template.format(*line)} ," for line in lines),
+                    "]",
+                ]
+            )
```

Comment:
The Black and Ruff formatter styles will be updated at some point early 2025. When that happens [this will be rewritten as](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-multiline-dictionary-and-list-indentation-for-sole-function-parameter)
```suggestion
            return "\n".join([
                row_template.format(*self.HEADING_NAMES),
                "[",
                *(f"{row_template.format(*line)} ," for line in lines),
                "]",
            ])
```
You can already try this out locally with `ruff format --preview --diff astropy/units/core.py`.